ui/app/selectors/permissions.js

```diff
@@ -1,48 +1,130 @@
+import {
+  createSelectorCreator,
+  defaultMemoize,
+} from 'reselect'
+import { isEqual } from 'lodash'
 
-import { createSelector } from 'reselect'
 import {
   CAVEAT_NAMES,
 } from '../../../app/scripts/controllers/permissions/enums'
 
-const permissionsSelector = (state, origin) => {
-  return origin && state.metamask.domains && state.metamask.domains[origin]
-}
+// selector creators
 
-// all permissions for the origin probably too expensive for deep equality check
-const accountsPermissionSelector = createSelector(
-  permissionsSelector,
-  (domain = {}) => {
+const createAllDomainAccountsSelector = createSelectorCreator(
```

Comment:
I'm having a bit of trouble following what this custom selector creator is for, and what `allDomainsEqualByAccounts` means, and these related functions

Reply:
Oh I see - it's to avoid re-computing things if the keys don't change, since only the keys are used?

I think it would be simpler to move `Object.keys(domains)` into a separate selector - that would be a bit more idiomatic and easier to follow, and would achieve the same result of not computing if they stay the same

Reply:
Oh, it does a deep comparison. Then yeah, this selector creator seems entirely unnecessary